Feature/block27 Provision AKS environment pipeline (#39)
devonfw/hangar: documentation/azure-devops/setup-aks-provisioning-pipeline.asciidoc

-n, --pipeline-name      [Required] Name that will be set to the pipeline.
    --resource-group     [Required] Name of the resource group for the cluster.
    --storage--account   [Required] Name of the storage account for the cluster.
    --storage-container  [Required] Name of the storage container where the tfstate file of the cluster will be stored.
-d, --local-directory    [Required] Local directory of your project (the path should always be using '/' and not '\').
-b, --target-branch                 Name of the branch to which the Pull Request will target. PR is not created if the flag is not provided.
-w                                  Open the Pull Request on the web browser if it cannot be automatically merged. Requires -b flag.
```

=== Example

```
./pipeline_generator.sh -c ./templates/aks/aks-pipeline.cfg -n aks-provisioning --resource-group hangar-aks-rg --storage-account hangarstatefiles --storage-container tfstatefile -d C:/Users/$USERNAME/Desktop/quarkus-project -b develop -w
```

=== Appendix: Interacting with the cluster

NOTE: Make sure to have https://kubernetes.io/docs/tasks/tools/[kubectl] installed in order to interact with your cluster.

> In order to interact with your cluster download the artifact `kubeconfig` generated by the pipeline and execute the following command:

Reviewer: ```suggestion
In order to interact with your cluster you will need to download the artifact `kubeconfig` generated by the cluster provisioning pipeline on the location it is expected by default (`~/.kube/config`) or either:
```

```

> kubectl <kubectl command> --kubeconfig <path to downloaded kubeconfig>

Reviewer: ```suggestion
# via environment variable (you can add this on your profile)
export KUBECONFIG=<path to downloaded kubeconfig>
kubectl <command>

# via command-line flag
kubectl <command> --kubeconfig=<path to downloaded kubeconfig>
```

```
az group create -n <resource group name> -l <location>
```

* An Azure storage account within the previous resource group. You can use an existing one or create a new one with the following command:

```
az storage account create -n <storage account name> -g <resource group name> -l <location>
```

* An Azure storage container in Azure within the previous storage account. You can use an existing one or create a new one with the following command:

```
az storage container create -n <storage container name> --account-name <storage account name>
```

* This script will commit and push the corresponding YAML template into your repository, so please be sure your local repository is up-to-date (i.e you have pulled the latest changes with `git pull`).

== Creating the pipeline using provided script

Before executing the script you will need to customize some input variables about the environment. To do so, you can either edit `terraform.tfvars` file or take advantage of the `set-config.sh` script located at `/scripts/environment-provisioning/azure/aks`, which allows you to create or update values for the required variables, passing them as flags. As a full example:

```

> ./set-config.sh --cluster_name <cluster name> --location <location> --resource_group_name <resource group name> --instance_type <worker instance type> --worker_node_count <number of worker nodes> --dns_prefix <dns prefix>

Reviewer: ```suggestion
./set-terraform-variables.sh --cluster_name <cluster name> --location <location> --resource_group_name <resource group name> --instance_type <worker instance type> --worker_node_count <number of worker nodes> --dns_prefix <dns prefix>
```
